app/lib/activitypub/activity/follow.rb

```diff
@@ -6,7 +6,13 @@ def perform
 
     return if target_account.nil? || !target_account.local? || delete_arrived_first?(@json['id'])
 
-    follow = @account.follow!(target_account)
-    NotifyService.new.call(target_account, follow)
+    follow_request = FollowRequest.create!(account: @account, target_account: target_account)
+
+    if target_account.locked?
```

Comment:
does activitypub have a concept of a locked account? I'm not sure I follow the flow of this logic.